.github/update-published-charms-tests-workflow.py

```diff
@@ -0,0 +1,144 @@
+#! /usr/bin/env python
+
+# /// script
```

Comment:
Can this be under the Copyright header? Seems like that should be as close to the top as possible, for consistency.

Reply:
How do you run this script? Do we want a Tox command, or just a comment here with the `uv` or whatever incantation?

Reply:
> Can this be under the Copyright header? Seems like that should be as close to the top as possible, for consistency.

Sure. I haven't seen these used much yet (now that both pipx and uv have support I imagine they'll get more common), so wasn't really sure where the right placing was. I can't see any [ordering instructions in the docs](https://packaging.python.org/en/latest/specifications/inline-script-metadata/#inline-script-metadata) - it seems like maybe you can put the comment anywhere as long as it's top-level. I'll move the copyright above.

> How do you run this script? Do we want a Tox command, or just a comment here with the `uv` or whatever incantation?

I simplified it to only use `pyyaml` and the standard lib so if you're in an environment with the ops dependencies then you can just `python update-published-charms-tests-workflow.py`. If you're not then using `uv run` will handle getting pyyaml, and I think maybe pipx can be used too (but I'm not sure what the incantation is for a single-file run).

I can add in instructions, or if it's preferred I can drop the `pyyaml` requirement and instead of parsing the YAML and then dumping it out again, I can just open it as text and do a basic text replacement. I have to do that once anyway because of the 'on:' to 'true:' quirk. Any preference?

Reply:
I think we should do basic text replacement, but not really for the dependencies reason -- but because it will solve the `run` formatting shenanigans, too.